client/web/src/enterprise/campaigns/detail/changesets/ExternalChangesetInfoCell.tsx

```diff
@@ -85,6 +90,13 @@ function isImporting(node: ExternalChangesetFields): boolean {
     )
 }
 
+function headReference(node: ExternalChangesetFields): string | undefined {
+    if (!isImporting(node) && node.currentSpec?.description.__typename === 'GitBranchChangesetDescription') {
+        return node.currentSpec?.description.headRef
+    }
+    return undefined
+}
```

Comment:
If we did something like
```suggestion
function hasHeadReference(node: ExternalChangesetFields): node is ExternalChangesetFields & { currentSpec: ChangesetSpecFields & { description: GitBranchChangesetDescriptionFields  } } {
return node.currentSpec?.description.__typename === 'GitBranchChangesetDescription'
}
```
we'd be able to simplify this above and wouldn't have to think about if there is the potential to have an empty badge displayed :)
(Hint: the type is probably wrong, but something similar should work)

Reply:
Guards! Guards!

For my own interest, could the badge just be a tiny component of its own that doesn't render anything if there's no `headRef`? Would that be cleaner?

Reply:
Thanks, I used your function. Sadly it doesn't work as a type guard in the JSX snippet, so I need to check twice, I think.